eclipse-che/che-operator
# feat: sync Secrets, ConfigMaps and PersistentVolumesClaims to users n…

…amespaces



### What does this PR do?
sync Secrets, ConfigMaps and PersistentVolumesClaims to users namespases

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20501

### How to test this PR?


1. Prepare a patch file if needed:
```bash
cat > /tmp/cr-patch.yaml <<EOF
apiVersion: org.eclipse.che/v2
kind: CheCluster
spec: {}
EOF
```

2. Deploy the operator:
 
#### OpenShift
```bash
./build/scripts/olm/test-catalog-from-sources.sh --cr-patch-yaml /tmp/cr-patch.yaml
```

#### on Minikube

```bash
./build/scripts/minikube-tests/test-operator-from-sources.sh --cr-patch-yaml /tmp/cr-patch.yaml
```
3. Create 500 namespaces
```
for i in {1..500}; do                                          
  oc apply -f - <<EOF
kind: Namespace
apiVersion: v1
metadata:
  name: test-$i
  labels:
    app.kubernetes.io/component: "workspaces-namespace"
    app.kubernetes.io/part-of: "che.eclipse.org"
    che.eclipse.org/che-namespace: "test-$i"
    che.eclipse.org/che-name: "che"
    che.eclipse.org/workspace-namespace-owner-uid: "owner-id"
  annotations:
    che.eclipse.org/username: "test-$i"
EOF
done
```

4. Create a ConfigMap in eclipse-che namespace
```
oc apply -f - <<EOF                                            
kind: ConfigMap   
apiVersion: v1
metadata:
  name: dev-env-variables
  namespace: eclipse-che
  labels:
    app.kubernetes.io/part-of: che.eclipse.org
    app.kubernetes.io/component: workspaces-config
EOF
```
5. Wait until reconcile is completed
```
oc logs deploy/che-operator -n eclipse-che -f
```

6. Check that CM exists in every namespace
```
for i in {1..500}; do                                  
  oc get cm -n test-$i dev-env-variables                 
done
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
